Lib/test/test_concurrent_futures.py

```diff
@@ -303,6 +303,33 @@ def test_interpreter_shutdown(self):
         self.assertFalse(err)
         self.assertEqual(out.strip(), b"apple")
 
+    def test_submit_after_interpreter_shutdown(self):
+        # Test the atexit hook for shutdown of worker threads and processes
+        rc, out, err = assert_python_ok('-c', """if 1:
+            import atexit
+            @atexit.register
+            def run_last():
+                try:
+                    t.submit(lambda: None)
+                except RuntimeError:
+                    print("runtime-error")
+                    raise
+            from concurrent.futures import {executor_type}
+            if __name__ == "__main__":
+                context = '{context}'
+                if context == "":
+                    t = {executor_type}(5)
+                else:
+                    from multiprocessing import get_context
+                    context = get_context(context)
+                    t = {executor_type}(5, mp_context=context)
```

Comment:
Here, the executor is not started before the call in `atexit`. This causes another set of problem as the `sub{Process,Thread}` are not started and cleaned properly. For your proposed test, I would add a `t.submit(id, 42).result()` to be sure that the executor has been started before the atexit calls.

Reply:
Added `t.submit(id, 42).result()` to the test in d5c28194a0b842833ac4729f5f333ce718120201